scrapy/http/request/json_request.py

```diff
@@ -0,0 +1,31 @@
+"""
+This module implements the JSONRequest class which is a more convenient class
+(than Request) to generate JSON Requests.
+
+See documentation in docs/topics/request-response.rst
+"""
+
+import json
+import warnings
+
+from scrapy.http.request import Request
+
+
+class JSONRequest(Request):
+    def __init__(self, *args, **kwargs):
+        body_passed = kwargs.get('body', None) is not None
+        data = kwargs.pop('data', None)
+        data_passed = data is not None
+
+        if body_passed and data_passed:
+            warnings.warn('Both body and data passed. data will be ignored')
+
+        elif not body_passed and data_passed:
+            kwargs['body'] = json.dumps(data, sort_keys=True)
```

Comment:
There are other kwargs which can be passed to dumps function, most importantly - ensure_ascii=False; without it some request bodies can be 2x larger than needed. What do you think about adding `dumps_kwargs` parameter, which allows to override keyword arguments passed to json.dumps (including ensure_ascii and sort_keys)?

I think it can also be a good idea to move serialization to a method; it would make it easier to e.g. make a subclass which uses ujson instead of stdlib json library.

Reply:
Agree. Changed.